### PR TITLE
i18n(es): add api reference

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### What kind of changes does this PR include?

- Translated content

#### Description

- adds `refenrence/api-reference.md` in spanish
- updates a11y titles in `ui.ts`
- add page to `nav.ts`

### CODE REVIEW

File: src/pages/es/reference/api-reference.md [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+
+:::caution
+La función `getStaticPaths()` se ejecuta en su propio ámbito aislado una vez, antes de que se cargue cualquier página. Por lo tanto, no puede hacer referencia a nada desde el ámbito principal, aparte de las importaciones de archivos. El compilador le advertirá si incumple este requisito.
+:::
+
+### `params`
+
+La key `params` de cada objeto devuelto le dice a Astro qué rutas construir. Los parámetros devueltos deben corresponder con los parámetros dinámicos y los parámetros comodín definidos en la ruta de archivo de su componente.
+
+Los `params` están codificados en la URL, por lo que solo se admiten cadenas y números como valores. El valor de cada objeto `params` debe coincidir con los parámetros utilizados en el nombre de la página.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:286]:
```suggestion
Los `params` están codificados en la URL, por lo que solo se admiten strings y números como valores. El valor de cada objeto `params` debe coincidir con los parámetros utilizados en el nombre de la página.
```

File: src/pages/es/reference/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+
+Por ejemplo, supongamos que tienes una página en `src/pages/posts/[id].astro`. Si exporta `getStaticPaths` desde esta página y devuelve lo siguiente para las rutas:
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:288]:
```suggestion
Por ejemplo, supongamos que tienes una página en `src/pages/posts/[id].astro`. Si exportas `getStaticPaths` desde esta página y devuelves lo siguiente para las rutas:
```

File: src/pages/es/reference/api-reference.md [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+const { id } = Astro.params;
+---
+<h1>{id}</h1>
+```
+
+Luego, Astro generará estáticamente `posts/1`, `posts/2` y `posts/3` en el momento de la compilación.
+
+### Transferencia de datos con `props`
+
+Para pasar datos adicionales a cada página generada, también puede establecer un valor `props` en cada objeto de ruta devuelto. A diferencia de `params`, `props` no están codificados en la URL y, por lo tanto, no están limitados a solo cadenas.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:309]:
```suggestion
Para pasar datos adicionales a cada página generada, también puedes establecer un valor `props` en cada objeto de ruta devuelto. A diferencia de `params`, `props` no están codificados en la URL y, por lo tanto, no están limitados a solamente strings.
```

File: src/pages/es/reference/api-reference.md [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+
+Por ejemplo, suponga que genera páginas basadas en datos obtenidos de una API remota. Puede pasar el objeto de datos completo al componente de página dentro de `getStaticPaths`:
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:311]:
```suggestion
Por ejemplo, suponiendo que generas páginas basadas en datos obtenidos de una API remota. Puedes pasar el objeto de datos completo al componente de página dentro de `getStaticPaths`:
```

File: src/pages/es/reference/api-reference.md [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+  <h1>{id}: {post.title}</h1>
+  <h2>Categoría: {post.category}</h2>
+</body>
+```
+
+Luego, Astro generará estáticamente `posts/1` y `posts/2` en el momento de la compilación usando el componente de página en `pages/posts/[id].astro`. La página puede hacer referencia a estos datos usando `Astro.props`:
+
+### `paginate()`
+
+La paginación es un caso de uso común para los sitios web que Astro admite de forma nativa a través de la función `paginate()`. `paginate()` generará automáticamente un array para devolver desde `getStaticPaths()` que creará una URL para cada página de la colección paginada. El número de página se pasará como un parámetro, y los datos de la página se pasarán como una prop de `página`.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:361]:
```suggestion
La paginación es un caso de uso común para los sitios web que Astro admite de forma nativa a través de la función `paginate()`. `paginate()` generará automáticamente un array para devolver desde `getStaticPaths()` que creará una URL para cada página de la colección paginada. El número de página se pasará como un parámetro, y los datos de la página se pasarán como una prop `page`.
```

File: src/pages/es/reference/api-reference.md [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+  // Devuelve una colección paginada de rutas para todas los artículos.
+  return paginate(allPokemon, { pageSize: 10 });
+}
+
+// Si se configuró correctamente, La prop de la página ahora tiene todo lo que
+// necesita para renderizar una página (vea la siguiente sección).
+const { page } = Astro.props;
+```
+
+`paginate()` asume un nombre de archivo de `[page].astro` o `[...page].astro`. El parámetro `page` se convierte en el número de página en la URL:
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:379]:
```suggestion
`paginate()` asume un nombre de archivo `[page].astro` o `[...page].astro`. El parámetro `page` se convierte en el número de página en la URL:
```

File: src/pages/es/reference/api-reference.md [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+
+- `/posts/[page].astro` generaría las URL `/posts/1`, `/posts/2`, `/posts/3`, etc.
+- `/posts/[...page].astro` generaría las URL `/posts`, `/posts/2`, `/posts/3`, etc.
+
+#### La prop de paginación `page`
+
+La paginación pasará una prop de `página` a cada página renderizada que represente una sola página de datos en la colección paginada. Esto incluye los datos que ha paginado (`page.data`), así como los metadatos de la página (`page.url`, `page.start`, `page.end`, `page.total`, etc.) . Estos metadatos son útiles para cosas como un botón "Página siguiente" o un mensaje "Mostrando 1-10 de 100".
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:386]:
```suggestion
La paginación pasará una prop `page` a cada página renderizada que represente una sola página de datos en la colección paginada. Esto incluye los datos que ha paginado (`page.data`), así como los metadatos de la página (`page.url`, `page.start`, `page.end`, `page.total`, etc.) . Estos metadatos son útiles para cosas como un botón "Página siguiente" o un mensaje "Mostrando 1-10 de 100".
```

File: src/pages/es/reference/api-reference.md [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+
+| Nombre             |         Tipo          | Descripción                                                                                                                       |
+| :----------------- | :-------------------: | :-------------------------------------------------------------------------------------------------------------------------------- |
+| `page.data`        |        `Array`        | Array de datos devueltos desde `data()` para la página actual.                                                                     |
+| `page.start`       |       `number`        | Índice del primer elemento en la página actual, comenzando en `0` (por ejemplo, si `pageSize: 25`, sería `0` en la página 1, `25` en la página 2, etc.). |
+| `page.end`         |       `number`        | Índice del último elemento de la página actual.                                                                                               |
+| `page.size`        |       `number`        | Cuantos articulos por página.                                                                   |
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:393]:
```suggestion
| `page.size`        |       `number`        | Cantidad de artículos por página.                                                                   |
```

File: src/pages/es/reference/api-reference.md [hunk 9]
```diff
@@ -0,0 +1,511 @@
...
+| `page.total`       |       `number`        | El número total de elementos en todas las páginas.                                                                                       |
+| `page.currentPage` |       `number`        | El número de página actual, comenzando con `1`.                                                                                       |
+| `page.lastPage`    |       `number`        | El número total de páginas.                                                                                                        |
+| `page.url.current` |       `string`        | La URL de la página actual (útil para URL canónicas)                                                                       |
+| `page.url.prev`    | `string \| undefined` | La URL de la página anterior (será `undefined` si está en la página 1).                                                              |
+| `page.url.next`    | `string \| undefined` | La URL de la siguiente página (será `indefinida` si no hay más páginas).                                                              |
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:399]:
```suggestion
| `page.url.next`    | `string \| undefined` | La URL de la siguiente página (será `undefined` si no hay más páginas).                                                              |
```

File: src/pages/es/reference/api-reference.md [hunk 10]
```diff
@@ -0,0 +1,511 @@
...
+
+## `import.meta`
+
+Todos los módulos ESM incluyen una propiedad `import.meta`. Astro agrega `import.meta.env` a través de [Vite](https://vitejs.dev/guide/env-and-mode.html).
+
+**`import.meta.env.SSR`** se puede usar para saber cuándo se renderiza en el servidor. A veces, es posible que desee una lógica diferente, por ejemplo, un componente que solo debe representarse en el cliente:
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:405]:
```suggestion
**`import.meta.env.SSR`** se puede usar para saber cuándo se renderiza en el servidor. A veces, es posible que desees una lógica diferente, por ejemplo, un componente que solo debe representarse en el cliente:
```

File: src/pages/es/reference/api-reference.md [hunk 11]
```diff
@@ -0,0 +1,511 @@
...
+import { h } from 'preact';
+
+export default function () {
+  return import.meta.env.SSR ? <div class="spinner"></div> : <FancyComponent />;
+}
+```
+
+## Componentes incorporados
+
+Astro incluye varios componentes incorporados para que los use en tus proyectos. Todos los componentes incorporados están disponibles en archivos `.astro` a través de `import {} from 'astro/components';`.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:417]:
```suggestion
Astro incluye varios componentes incorporados para que los uses en tus proyectos. Todos los componentes incorporados están disponibles en archivos `.astro` a través de `import {} from 'astro/components';`.
```

File: src/pages/es/reference/api-reference.md [hunk 12]
```diff
@@ -0,0 +1,511 @@
...
+
+### `<Markdown />`
+
+:::caution
+El componente `<Markdown />` no funciona en SSR y se moverá a su propio paquete antes de v1.0. Considere [importar contenido de Markdown](/es/guides/markdown-content/#importando-markdown) en su lugar.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:422]:
```suggestion
El componente `<Markdown />` no funciona en SSR y se moverá a su propio paquete antes de v1.0. Considera [importar contenido de Markdown](/es/guides/markdown-content/#importando-markdown) en su lugar.
```

File: src/pages/es/reference/api-reference.md [hunk 13]
```diff
@@ -0,0 +1,511 @@
...
+:::
+
+```astro
+---
+import { Markdown } from 'astro/components';
+---
+<Markdown>
+  # Markdown syntax is now supported! **Yay!**
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:430]:
```suggestion
  # ¡Aquí usamos sintaxis Markdown! **¡Yay!**
```

File: src/pages/es/reference/api-reference.md [hunk 14]
```diff
@@ -0,0 +1,511 @@
...
+</Markdown>
+```
+
+Consulte nuestra [guía de Markdown](/es/guides/markdown-content/) para obtener más información.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:434]:
```suggestion
Consulta nuestra [guía de Markdown](/es/guides/markdown-content/) para obtener más información.
```

File: src/pages/es/reference/api-reference.md [hunk 15]
```diff
@@ -0,0 +1,511 @@
...
+
+```astro
+---
+import { Code } from 'astro/components';
+---
+<!-- Resaltado de sintaxis de algún código JavaScript. -->
+<Code code={`const foo = 'bar';`} lang="js" />
+<!-- Opcional: personaliza tu tema. -->
+<Code code={`const foo = 'bar';`} lang="js" theme="dark-plus" />
+<!-- Opcional: habilite css wrap -->
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:448]:
```suggestion
<!-- Opcional: habilita css wrap -->
```

File: src/pages/es/reference/api-reference.md [hunk 16]
```diff
@@ -0,0 +1,511 @@
...
+<Code code={`const foo = 'bar';`} lang="js" wrap />
+```
+
+Este componente proporciona resaltado de sintaxis para bloques de código en el momento de la compilación (no incluye JavaScript del lado del cliente). El componente funciona internamente con Shiki y es compatible con todos los [temas populares](https://github.com/shikijs/shiki/blob/main/docs/themes.md) y [lenguajes de programación](https://github.com/shikijs/shiki/blob/main/docs/languages.md). Además, puedes agregar temas y lenguajes de programación personalizados modificando `theme` y `lang` respectivamente.
+
+### `<Prism />`
+
+:::note[Instalación]
+
+Para usar el componente resaltador `Prism`, primero **instale** el paquete `@astrojs/prism`:
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:458]:
```suggestion
Para usar el componente resaltador `Prism`, primero **instala** el paquete `@astrojs/prism`:
```

File: src/pages/es/reference/api-reference.md [hunk 17]
```diff
@@ -0,0 +1,511 @@
...
+:::
+
+```astro
+---
+import { Prism } from '@astrojs/prism';
+---
+<Prism lang="js" code={`const foo = 'bar';`} />
+```
+
+Este componente proporciona resaltado de sintaxis específico con el lenguaje de programación para bloques de código al aplicar las clases CSS de Prism. Ten en cuenta que **debes proporcionar una hoja de estilo Prism CSS** (o traer la tuya propia) para que aparezca el resaltado de sintaxis. Consulte la [sección de configuración de Prism](/es/guides/markdown-content/#configuración-de-prism) para obtener más detalles.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:489]:
```suggestion
Este componente proporciona resaltado de sintaxis específico con el lenguaje de programación para bloques de código al aplicar las clases CSS de Prism. Ten en cuenta que **debes proporcionar una hoja de estilo Prism CSS** (o traer la tuya propia) para que aparezca el resaltado de sintaxis. Consulta la [sección de configuración de Prism](/es/guides/markdown-content/#configuración-de-prism) para obtener más detalles.
```

File: src/pages/es/reference/api-reference.md [hunk 18]
```diff
@@ -0,0 +1,511 @@
...
+
+Consulte la [lista de idiomas admitidos por Prism](https://prismjs.com/#supported-languages) donde puede encontrar el alias correspondiente de un idioma. ¡Y también puede mostrar tus bloques de código Astro con `lang="astro"`!
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:491]:
```suggestion
Consulta la [lista de idiomas admitidos por Prism](https://prismjs.com/#supported-languages) donde puedes encontrar el alias correspondiente de un idioma. ¡Y también puedes mostrar tus bloques de código Astro con `lang="astro"`!
```

**Review by agustinmulet — COMMENTED** [review]
Reviewed, sorry for taking so long 🥲

**Comment by kevinzunigacuellar** [comment]
LGTM!

**Comment by Yan-Thomas** [comment]
Nice work @kevinzunigacuellar & @agustinmulet, thanks so much for this! There were some new changes: https://github.com/withastro/docs/commit/92666e4853eb21770c4cf07d732fcc8d65c61fff and https://github.com/withastro/docs/commit/c4bb2447c7bb5450bd04cea7070a1498d399cef2. After adding these I will review and merge the PR. 🙌

**Comment by kevinzunigacuellar** [comment]
> Nice work @kevinzunigacuellar & @agustinmulet, thanks so much for this! There were some new changes: [92666e4](https://github.com/withastro/docs/commit/92666e4853eb21770c4cf07d732fcc8d65c61fff) and [c4bb244](https://github.com/withastro/docs/commit/c4bb2447c7bb5450bd04cea7070a1498d399cef2). After adding these I will review and merge the PR. 🙌

thanks for caching those

File: src/pages/es/reference/api-reference.md [hunk 19]
```diff
@@ -0,0 +1,517 @@
...
+```astro
+---
+Astro.response.headers.set('Set-Cookie', 'a=b; Path=/;');
+---
+```
+
+### `Astro.canonicalURL`
+
+:::caution[Obsoleto]
+Utilice [`Astro.url`](#astrourl) para construir su propia URL canónica.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:134]:
```suggestion
Utiliza [`Astro.url`](#astrourl) para construir tu propia URL canónica.
```

File: src/pages/es/reference/api-reference.md [hunk 20]
```diff
@@ -0,0 +1,517 @@
...
+:::
+
+La [URL canónica][canonical] de la página actual.
+
+### `Astro.url`
+
+<Since v="1.0.0-rc" />
+
+Un objeto [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) construido a partir del valor actual de la string URL `Astro.request.url`. Útil para interactuar con propiedades individuales de la URL de la solicitud, como la ruta o el origen.
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:143]:
```suggestion
Un objeto [URL](https://developer.mozilla.org/es/docs/Web/API/URL) construido a partir del valor actual de la string URL `Astro.request.url`. Útil para interactuar con propiedades individuales de la URL de la solicitud, como la ruta o el origen.
```

File: src/pages/es/reference/api-reference.md [hunk 21]
```diff
@@ -0,0 +1,517 @@
...
+
+Equivalente a hacer `new URL (Astro.request.url)`.
+
+```astro
+<h1>La URL actual es: {Astro.url}</h1>
+<h1>El nombre de la ruta URL actual es: {Astro.url.pathname}</h1>
+<h1>El origen de la URL actual es: {Astro.url.origin}</h1>
+```
+
+También puedes usar `Astro.url` para crear nuevas URL pasándola como argumento a [`new URL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL).
```
Comment (agustinmulet) [inline src/pages/es/reference/api-reference.md:153]:
```suggestion
También puedes usar `Astro.url` para crear nuevas URL pasándola como argumento a [`new URL()`](https://developer.mozilla.org/es/docs/Web/API/URL/URL).
```

**Review by agustinmulet — COMMENTED** [review]
Reviewed! 💪

**Comment by kevinzunigacuellar** [comment]
thank you, everyone! let's merge this before another change tries to slide under the hood 😅

**Comment by Yan-Thomas** [comment]
> thank you, everyone! let's merge this before another change tries to slide under the hood 😅thank you, everyone! let's merge this before another change tries to slide under the hood 😅

Totally agree. Thanks, everyone! 🚀

**Pull request merged**